src/veil/ringct/anonwallet.cpp

```diff
@@ -387,12 +387,106 @@ bool AnonWallet::GetAddressMeta(const CStealthAddress& address, CKeyID& idAccoun
     if (mapKeyPaths.count(sxAddr.GetID())) {
         auto accountpair = mapKeyPaths.at(sxAddr.GetID());
         idAccount = accountpair.first;
+        std::string strPathToKey = BIP32PathToString(accountpair.second);
+
+        //The path of the account is what provides context
+        accountpair = mapKeyPaths.at(idAccount);
```

Comment:
How do we know that the mapKeyPaths contains the idAccount?. Should do a check to make sure the map contains it before accessing the map.